catalog/imaging/imaging-plugin-nitf/src/main/java/org/codice/alliance/plugin/nitf/NitfPostIngestPlugin.java

```diff
@@ -60,28 +72,19 @@
 import org.slf4j.LoggerFactory;
 
 /**
- * This pre-storage plugin creates and stores the NITF thumbnail and NITF overview images. The
+ * This post-ingest plugin creates and stores the NITF thumbnail and NITF overview images. The
  * thumbnail is stored with the Metacard while the overview is stored in the content store.
  */
-public class NitfPreStoragePlugin implements PreCreateStoragePlugin, PreUpdateStoragePlugin {
+public class NitfPostIngestPlugin implements PostIngestPlugin {
 
-  private static final String IMAGE_NITF = "image/nitf";
-
-  static final MimeType NITF_MIME_TYPE;
-
-  static {
-    try {
-      NITF_MIME_TYPE = new MimeType(IMAGE_NITF);
-    } catch (MimeTypeParseException e) {
-      throw new ExceptionInInitializerError(
-          String.format("Unable to create MimeType from '%s': %s", IMAGE_NITF, e.getMessage()));
-    }
-  }
+  static final String IMAGE_NITF = "image/nitf";
 
   private static final String IMAGE_JPEG = "image/jpeg";
 
   private static final String IMAGE_JPEG2K = "image/jp2";
 
+  private static final String NITF_PROCESSING_KEY = "NitfPostIngestPlugin.Processing";
```

Comment:
What is this flag used for? Why do we check it in the `shouldGenerateContentItems` method? Does is ever get removed from (or set to false in) the metacard properties? It looks like it gets put in the metacard properties after the nitf process has been completed but before the update storage request. Should it be `"NitfPostIngestPlugin.Processed"` instead?

Reply:
This property is added to the request so that this plugin doesn't run again on the same data when the update storage request is sent. This is similar to a flag in the async framework. I will change NitfPostIngestPlugin.Processing to NitfPostIngestPlugin.Processed

Reply:
That makes sense. Maybe a comment near the definition would be worthwhile.